tools/pkglint/lib/rules.ts

```diff
@@ -46,9 +46,38 @@ export class DescriptionIsRequired extends ValidationRule {
   }
 }
 
+/**
+ * Verify that all packages have a publishConfig with a publish tag set.
+ */
+export class PublishConfigTagIsRequired extends ValidationRule {
+  public readonly name = 'package-info/publish-config-tag';
+
+  public validate(pkg: PackageJson): void {
+    if (pkg.json.private) { return; }
+
+    // While v2 is still under development, we publish all v2 packages with the 'next'
+    // distribution tag, while still tagging all v1 packages as 'latest'.
+    // The one exception is 'aws-cdk-lib', since it's a new package for v2.
+    const newV2Packages = ['aws-cdk-lib'];
+    const defaultPublishTag = (cdkMajorVersion() === 2 && !newV2Packages.includes(pkg.json.name)) ? 'next' : 'latest';
+
+    if (!pkg.json.publishConfig?.tag) {
```

Comment:
```suggestion
    if (pkg.json.publishConfig?.tag !== defaultPublishTag) {
```

We need this for new packages to be correctly fixed up during `yarn pkglint` during forward merges.